utils/req-res-log-valiadtor.py

```diff
@@ -0,0 +1,173 @@
+            for i in range(count):
+                field = Response(f)
+                assert isinstance(field.json, str)
+                value = Response(f)
+                self.json[str(field)[1:-1]] = value.json
+
+    def __str__(self):
+        return json.dumps(self.json)
+
+        
+# Figure out where the sources are
+srcdir = os.path.abspath(os.path.dirname(os.path.abspath(__file__)) + "/../src")
+testdir = os.path.abspath(os.path.dirname(os.path.abspath(__file__)) + "/../tests")
+
+if __name__ == '__main__':
+    print('Starting Redis server')
+    redis_proc = subprocess.Popen(['%s/redis-server' % srcdir, '--port', '6534'], stdout=subprocess.PIPE)
+    
+    while True:
+        try:
+            print('Connecting to Redis...')
+            r = redis.Redis(port=6534)
+            r.ping()
+            break
+        except Exception as e:
+            time.sleep(0.1)
+            pass
+
+    cli_proc = subprocess.Popen(['%s/redis-cli' % srcdir, '-p', '6534', '--json', 'command', 'docs'], stdout=subprocess.PIPE)
+    stdout, stderr = cli_proc.communicate()
+    docs = json.loads(stdout)
+
+    redis_proc.terminate()
+    redis_proc.wait()
+
+    # Create all command objects
+    missing_schema = set()
+    print("Processing files...")
+    for filename in glob.glob('%s/tmp/*/*.reqres' % testdir):
+        lineno = 0
+        with open(filename, "r", newline="\r\n", encoding="latin-1") as f:
+            print("Processing %s ..." % filename)
+            while True:
+                try:
+                    req = Request(f, docs)
+                    if not req.command:
+                        break
+                    res = Response(f)
+                except json.decoder.JSONDecodeError as err:
+                   print("JSON decoder error while processing %s:%d: %s" % (filename, i, err))
+                   exit(1)
+                except Exception as err:
+                   print("General error while processing %s:%d: %s" % (filename, lineno, err))
+                   exit(1)
+
+                if res.error or res.queued:
+                    continue
+
+                if not req.schema:
+                    missing_schema.add(req.command)
+                    continue
+
+                try:
+                    jsonschema.validate(instance=res.json, schema=req.schema)
+                except jsonschema.ValidationError as err:
+                    print("JSON schema validation error on %s: %s" % (filename, err))
+                    print("Command: %s" % req.command)
+                    try:
+                        print("Response: %s" % res)
+                    except UnicodeDecodeError as err:
+                       print("Response: (unprintable)")
+                    print("Schema: %s" % json.dumps(req.schema, indent=2))
+                    exit(1)
+        
+    print("Done.")
+    if missing_schema:
+        print("WARNING! The following commands are missing a reply_schema: {}".format(sorted(missing_schema)))
```

Comment:
let's add somewhere a list of validations that we do, and also include the planned ones that are yet to be implemented.
i.e.
* warn about the commands that are missing schema
* warn about about schemas that are uncovered by the tests output
* warn about replies that don't match the schema.

Reply:
1 and 3 are covered, added the 2nd point to the TODO in the top comment